domain/storage/types.go

```diff
@@ -20,13 +20,27 @@ const (
 // Attrs defines storage attributes.
 type Attrs map[string]string
 
+// StoragePoolOrigin defines the origin value of a storage pool.
+// It can be "user", "built-in", or "provider-default".
+type StoragePoolOrigin string
+
+const (
+	// StoragePoolOriginUser defines a storage pool created by a user.
+	StoragePoolOriginUser StoragePoolOrigin = "user"
+	// StoragePoolOriginBuiltIn defines a storage pool that is built-in to Juju.
+	StoragePoolOriginBuiltIn StoragePoolOrigin = "built-in"
+	// StoragePoolOriginProviderDefault defines a storage pool that is provided by the storage provider.
+	StoragePoolOriginProviderDefault StoragePoolOrigin = "provider-default"
+)
+
 // StoragePool represents a storage pool in Juju.
 // It contains the name of the pool, the provider type, and any attributes
 type StoragePool struct {
 	UUID     string
 	Name     string
 	Provider string
 	Attrs    Attrs
+	Origin   StoragePoolOrigin
```

Comment:
I don't think we need nor want this value at the moment.

As a side note I think structs like this tend to highlight a design problem sometimes. Because we overload the struct as both the creation args and the presentation value for a storage pool the insert logic now has to deal with origin and validating it.

If we had separated the concepts we don't have to chase our tail around with a value we don't want.